### PR TITLE
Feature/adj pos fr

### PR DESCRIPTION
**Pull request by blegaut**
Thanks for contributing!

## Please check if the PR fulfills these requirements 👌:

- [x] Confirm the PR related to a dedicated issue
- [x] Confirm your PR corrects a single bug or implements a single feature
- [x] Your code is linted locally prior to submission
- [x] Your code is fully tested: 99% to 100% coverage
- [ ] Everything is correct on [Sonar dashboard](https://sonarcloud.io/dashboard?id=RosaeNLG_RosaeNLG)
- [x] Confirm your code is under Apache 2.0 license
- [x] Confirm your documentation is under CC-BY-4.0 license
- [x] Confirm license and copyright content is created/updated in each file (see `CONTRIBUTING.md`)
- [x] Confirm `changelog.adoc` is updated
- [ ] If corrects vulnerabilities, confirm `changelog.adoc` contains CVE IDs

Also remember: each commit **MUST** contain a sign off message (see `CONTRIBUTING.md`) 🙄

## Indicate related issue: 

issue #...

### CODE REVIEW

**Comment by blegaut** [comment]
Not sure why the simple workflow tests are not running

File: packages/rosaenlg/src/LanguageFrench.ts [hunk 1]
```diff
@@ -140,24 +142,28 @@ export class LanguageFrench extends LanguageImpl {
     return frenchParse(val, { dictHelper: this.dictHelper });
   }
 
-  thirdPossessionTriggerRef(owner: any, owned: any, params: any): void {
+  thirdPossessionTriggerRef(owner: any, owned: any, params: ValueParams): void {
```
Comment (ludans) [inline packages/rosaenlg/src/LanguageFrench.ts:145]: One possible parameter is, in English, `possForm`: can be `S` or `OF`. But it is not defined is `ValueParams`. Can you please check?
- either add `possForm` to `ValueParams`, if it makes sense
- or create a new type

File: packages/rosaenlg/test/test-rosaenlg/fr_FR/value.pug [hunk 2]
```diff
@@ -151,3 +160,12 @@ t
 
   //- debug
   l #[+value('alsacien', {adj:'vieux', adjPos:'BEFORE', number:'P' } )]
+
+  //- adjectifs possessifs
```
Comment (ludans) [inline packages/rosaenlg/test/test-rosaenlg/fr_FR/value.pug:164]: a for loop or a eachz could make the code a little more compact

File: packages/rosaenlg/src/PossessiveManager.ts [hunk 3]
```diff
@@ -39,7 +40,7 @@ export class PossessiveManager {
     this.languageImpl.recipientPossession(owned);
   }
 
-  public thirdPossession(owner: any, owned: any, params: any): void {
+  public thirdPossession(owner: any, owned: any, params: ValueParams): void {
```
Comment (ludans) [inline packages/rosaenlg/src/PossessiveManager.ts:43]: same comment as top

File: packages/rosaenlg/src/LanguageGerman.ts [hunk 4]
```diff
@@ -134,6 +134,7 @@ export class LanguageGerman extends LanguageImpl {
       numberOwner: this.genderNumberManager.getRefNumber(owner, params),
       genderOwned: this.genderNumberManager.getRefGender(owned, params),
       numberOwned: this.genderNumberManager.getRefNumber(owned, params),
+      personOwner: 3, //first and second person not implemented
```
Comment (ludans) [inline packages/rosaenlg/src/LanguageGerman.ts:137]: I think it is not useful:
- the parameter is optional
- `getDet` implementation in `LanguageGerman.ts` does not use it

File: packages/rosaenlg/src/LanguageEnglish.ts [hunk 5]
```diff
@@ -144,6 +144,7 @@ export class LanguageEnglish extends LanguageImpl {
       genderOwner: this.genderNumberManager.getRefGender(owner, params),
       numberOwner: this.genderNumberManager.getRefNumber(owner, params),
       numberOwned: null, // we do not care
+      personOwner: 3, //first and second person not implemented
```
Comment (ludans) [inline packages/rosaenlg/src/LanguageEnglish.ts:147]: not really useful as parameter is optional and `getDet` implemtation in `LanguageEnglish.ts` does not use it

File: packages/rosaenlg/src/LanguageFrench.ts [hunk 6]
```diff
@@ -75,16 +75,18 @@ export class LanguageFrench extends LanguageImpl {
...
-      params.forceDes,
-    );
+  getDet(detType: DetTypes, params: DetParams): string {
+    const { genderOwned, numberOwned, numberOwner, adjectiveAfterDet, after, forceDes, personOwner } = params;
+    return getFrenchDet({
+      detType,
+      genderOwned: genderOwned as GendersMF,
+      numberOwned: numberOwned || 'S',
+      numberOwner: numberOwner || 'S',
+      personOwner: personOwner || 3,
```
Comment (ludans) [inline packages/rosaenlg/src/LanguageFrench.ts:85]: ok default value is set here

File: packages/rosaenlg/src/LanguageFrench.ts [hunk 7]
```diff
@@ -140,24 +142,28 @@ export class LanguageFrench extends LanguageImpl {
...
   }
 
-  thirdPossessionRefTriggered(owner: any, owned: any, params: any): void {
+  thirdPossessionRefTriggered(owner: any, owned: any, params: ValueParams): void {
     const det: string = this.getDet('POSSESSIVE', {
       genderOwned: this.genderNumberManager.getRefGender(owned, null),
       genderOwner: null,
       numberOwner: this.genderNumberManager.getRefNumber(owner, params),
       numberOwned: this.genderNumberManager.getRefNumber(owned, params),
+      personOwner: (params && params.personOwner) || 3,
```
Comment (ludans) [inline packages/rosaenlg/src/LanguageFrench.ts:157]: I would remove || 3
as the default value is already managed later

File: packages/rosaenlg-doc/doc/modules/ROOT/pages/changelog.adoc [hunk 8]
```diff
@@ -19,6 +19,8 @@ https://keepachangelog.com/en/0.3.0/
 
 === Added
 
+* in `value` mixin, ability to generate a possesive determiner, using `personOwner: 1`, e.g. to generate _ma maison_, _ta maison_ in French. The `personOwner` parameter is also available in the `sentence` mixin subject and object groups.
```
Comment (ludans) [inline packages/rosaenlg-doc/doc/modules/ROOT/pages/changelog.adoc:22]: please indicate more clearly that `personOwner` only works in French (for both `value` and `sentence` mixins)

**Review by ludans — CHANGES_REQUESTED** [review]
Thanks for the PR.

In `french-determiner` package:
- please update `example.js` (must be runnable when doing just `node example.js`)
- please update `README.md`: new parameters, example, etc.

The example with `sentence` in French is very good. Can you put it in the documentation, in `sentences_french.adoc`?

Can you please update the `value.adoc` page, and put an example in French?

Also, can you please rebase on master? The code structure has somewhat changed due to separating the packages.

Let's finish this one and do a new one for the English support of possessives.

File: packages/rosaenlg/src/LanguageGerman.ts (hunk 4 above)
Comment (blegaut) [reply packages/rosaenlg/src/LanguageGerman.ts:137]: ok, I ake it optional in DetParams

File: packages/rosaenlg/src/LanguageFrench.ts (hunk 1 above)
Comment (blegaut) [reply packages/rosaenlg/src/LanguageFrench.ts:145]: let's add it ot ValueParams

File: packages/rosaenlg/test/test-rosaenlg/fr_FR/value.pug (hunk 2 above)
Comment (blegaut) [reply packages/rosaenlg/test/test-rosaenlg/fr_FR/value.pug:164]: done

File: packages/rosaenlg-doc/doc/modules/ROOT/pages/changelog.adoc (hunk 8 above)
Comment (blegaut) [reply packages/rosaenlg-doc/doc/modules/ROOT/pages/changelog.adoc:22]: ok

File: packages/rosaenlg/src/LanguageFrench.ts [hunk 9]
```diff
@@ -140,24 +142,28 @@ export class LanguageFrench extends LanguageImpl {
...
 
-  thirdPossessionTriggerRef(owner: any, owned: any, params: any): void {
+  thirdPossessionTriggerRef(owner: any, owned: any, params: ValueParams): void {
     this.valueManager.value(owned, Object.assign({}, params, { det: 'DEFINITE' }));
     this.spy.appendPugHtml(` de `);
     this.valueManager.value(owner, Object.assign({}, params));
   }
 
-  thirdPossessionRefTriggered(owner: any, owned: any, params: any): void {
+  thirdPossessionRefTriggered(owner: any, owned: any, params: ValueParams): void {
```
[the remark this answers is not in the document]
Comment (blegaut) [reply packages/rosaenlg/src/LanguageFrench.ts]: ok

File: packages/rosaenlg/src/PossessiveManager.ts (hunk 3 above)
Comment (blegaut) [reply packages/rosaenlg/src/PossessiveManager.ts]: ok

File: packages/rosaenlg/src/LanguageEnglish.ts (hunk 5 above)
Comment (blegaut) [reply packages/rosaenlg/src/LanguageEnglish.ts]: removed

File: packages/rosaenlg/src/LanguageFrench.ts (hunk 7 above)
Comment (blegaut) [reply packages/rosaenlg/src/LanguageFrench.ts]: removed

**Comment by blegaut** [comment]
> Thanks for the PR.
> 
> In `french-determiner` package:
> 
> * please update `example.js` (must be runnable when doing just `node example.js`)
> * please update `README.md`: new parameters, example, etc.
> 
> The example with `sentence` in French is very good. Can you put it in the documentation, in `sentences_french.adoc`?
> 
> Can you please update the `value.adoc` page, and put an example in French?
> 
> Also, can you please rebase on master? The code structure has somewhat changed due to separating the packages.
> 
> Let's finish this one and do a new one for the English support of possessives.

Everything should be ok. Even If I am not clear how to test the doc

**Review by ludans — APPROVED** [review]

**Comment by ludans** [comment]
Perfect, thanks, I will merge.

To test the code within the documentation:
1. In the doc, use this structure:

++++
<script>
spawnEditor('language e.g. fr_FR', 
`
some standalone RosaeNLG scripting
`, 'full expected text, or fragment of expected text (will test that result contains this text)'
);
</script>
++++

2. In `rosaenlg-lib` package, run: `mocha test/pug/codeInDoc.js`
It will parse the doc for code, run the code, check that it runs and that the result is correct.

(I'll put this in the developper documentation)

**Pull request merged**